plugins/reachability/src/main/java/org/qbicc/plugin/reachability/ReachabilityBlockBuilder.java

```diff
@@ -202,150 +199,162 @@ public Value classOf(Value typeId) {
         return super.classOf(typeId);
     }
 
-    void enqueueClassInitializers(final LoadedTypeDefinition ltd) {
-        if (ltd.isInterface()) {
-            // a static field access or static method call on an interface
-            // only enqueues the interface's <clinit>, not it's super class
-            // or interfaces
+    private void processClassInitialization(final LoadedTypeDefinition ltd) {
+        Assert.assertFalse(ltd.isInterface());
+        if (info.isInitializedType(ltd)) return;
+
+        if (ltd.hasSuperClass()) {
+            // force superclass initialization
+            processClassInitialization(ltd.getSuperClass());
+        }
+        info.addInitializedType(ltd);
+
+        if (ltd.getInitializer() != null) {
             ctxt.enqueue(ltd.getInitializer());
-        } else {
-            // For a class, walk the class heirarchy and enqueue the all
-            // <clinit> in the superclass chain
-            LoadedTypeDefinition iterator = ltd;
-            while (iterator != null) {
-                InitializerElement initializer = iterator.getInitializer();
-                if (initializer != null) {
-                    ctxt.enqueue(initializer);
+        }
+
+        // Annoyingly, because an intermediate interface could be marked initialized due to a static field
+        // access which doesn't cause the initialization of its superinterfaces, we can't short-circuit
+        // this walk up the entire interface hierarchy when we hit an already initialized interfaces.
+        ArrayDeque<LoadedTypeDefinition> worklist = new ArrayDeque<>(List.of(ltd.getInterfaces()));
```

Comment:
I got tired of writing this code so I added a `void forEachInterfaceFullImplementedSet(Consumer<LoadedTypeDefinition> function);` helper to LoadedTypeDefinition.

I think this is equivalent to:
```java
ltd.forEachInterfaceFullImplementedSet( i -> { 
    if (i.declaresDefaultMethods() && !info.isInitializedType(i)) {
             info.addInitializedType(i);
             if (i.getInitializer() != null) {
                 ctxt.enqueue(i.getInitializer());
             }
     }
};
```

Feel free to ignore this suggestion

Reply:
I saw that method, put I didn't think it actually went up the superinterface hierarchy.  pretty sure LTD.getInterfaces() is only one level...not whole crawl up the interface extends hierarchy